apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -1049,11 +1049,8 @@ func (m *ModelManagerAPI) ModifyModelAccess(args params.ModifyModelAccessRequest
 			result.Results[i].Error = apiservererrors.ServerError(errors.Annotate(err, "could not modify model access"))
 			continue
 		}
-		err = m.authorizer.HasPermission(permission.AdminAccess, modelTag)
-		if err != nil {
-			return result, errors.Trace(err)
-		}
-		canModify := err == nil || canModifyController
+
+		canModify := m.authorizer.HasPermission(permission.AdminAccess, modelTag) == nil || canModifyController
```

Comment:
You are throwing an error away here. This needs to be checked properly using the same approach as above for canModifyController.

Please put another PR to fix this.